Makefile

```diff
@@ -0,0 +1,116 @@
+#Compiler and Linker
+CC          := g++
+
+#The Target Binary Program
+TARGET      := linapple
+
+#The Directories, Source, Includes, Objects, Binary and Resources
+SRCDIR      := src
+INCDIR      := inc
+BUILDDIR    := obj
+TARGETDIR   := bin
+RESDIR      := res
+SRCEXT      := cpp
+DEPEXT      := d
+OBJEXT      := o
+
+VERSION			:= 2.1.1
+PKG         := linapple
+
+INSTDIR = /usr/local/bin/$(EXE)
+CONFDIR = ~/$(TARGET)
+
+#Flags, Libraries and Includes
+
+SDL_CONFIG ?= sdl-config
+SDL_CFLAGS = $(shell $(SDL_CONFIG) --cflags)
+SDL_LIBS = $(shell $(SDL_CONFIG) --libs)
+
+CURL_CONFIG ?= curl-config
+CURL_CFLAGS = $(shell $(CURL_CONFIG) --cflags)
+CURL_LIBS = $(shell $(CURL_CONFIG) --libs)
+
+#PROFILING
+#CFLAGS      := -Wall -O0 -pg -ggdb -ansi -c
+#LFLAGS      := -pg
+#DEBUGGING
+#CFLAGS      := -Wall -O0 -ggdb -ansi -c -finstrument-functions
+#OPTIMIZED
+CFLAGS      := -Wall -O3 -ansi -c
+CFLAGS 		+= $(SDL_CFLAGS)
+CFLAGS 		+= $(CURL_CFLAGS)
+
+LIB 				:= $(SDL_LIBS) $(CURL_LIBS) -lz -lzip
+INC         := -I$(INCDIR) -I/usr/local/include
+INCDEP      := -I$(INCDIR)
+
+#---------------------------------------------------------------------------------
+#DO NOT EDIT BELOW THIS LINE
+#---------------------------------------------------------------------------------
+SOURCES     := $(shell find $(SRCDIR) -type f -name *.$(SRCEXT))
+OBJECTS     := $(patsubst $(SRCDIR)/%,$(BUILDDIR)/%,$(SOURCES:.$(SRCEXT)=.$(OBJEXT)))
+
+#Defauilt Make
+all: resources $(TARGET)
+
+#Remake
+remake: cleaner all
+
+#Copy Resources from Resources Directory to Target Directory
+resources: directories
+		@cp $(RESDIR)/* $(TARGETDIR)/
+
+#Make the Directories
+directories:
+		@mkdir -p $(TARGETDIR)
+		@mkdir -p $(BUILDDIR)
+
+package: all
+	@echo " Building a package"
+	mkdir -p "pkg/$(PKG)/$(PKG)-$(VERSION)/etc/$(PKG)"
+	mkdir -p "pkg/$(PKG)/$(PKG)-$(VERSION)/usr/bin/"
+	@cp -rf $(RESDIR)/* "pkg/$(PKG)/$(PKG)-$(VERSION)/etc/$(PKG)"
+	@cp $(TARGETDIR)/$(TARGET) "pkg/$(PKG)/$(PKG)-$(VERSION)/usr/bin/$(TARGET)"
+	chown -R root:root "pkg/$(PKG)"
+	dpkg --build "pkg/$(PKG)/$(PKG)-$(VERSION)"
+	mv "pkg/$(PKG)/$(PKG)-$(VERSION).deb" .
+
+install: all
+	@echo " o Creating install directory '$(INSTDIR)'"
+	mkdir -p "$(INSTDIR)"
+	chmod 777 "$(INSTDIR)"
+	@echo " o Creating additional directories 'conf' and 'ftp' in '$(INSTDIR)'"
+	mkdir "$(CONFDIR)/conf"
+	mkdir -p "$(CONFDIR)/sound"
+
+#Clean only Objecst
+clean:
+		@$(RM) -rf $(BUILDDIR)
+		@$(RM) -rf $(TARGETDIR)
+
+#Full Clean, Objects and Binaries
+cleaner: clean
+		@$(RM) -rf $(TARGETDIR)
+		@$(RM) $(TARGET)-$(VERSION).deb
+
+#Pull in dependency info for *existing* .o files
+-include $(OBJECTS:.$(OBJEXT)=.$(DEPEXT))
+
+#Link
+
+$(TARGET): $(OBJECTS)
+		$(CC) $(LFLAGS) -o $(TARGETDIR)/$(TARGET) $^ $(LIB)
```

Comment:
Are you opposed to autoconf or even automake?  Some people are fanatically opposed because of how ridiculously overcomplicated it is, particularly in terms of what it generates, in the name of compatibility with 60 year old UNIX systems nobody's using anymore.  I tried to love CMake, and I certainly love how it can generate all the project files for all the platforms, but it always felt like I was wrestling with it.  Maybe I'm just more familiar with autoconf and potentially automake.

Which isn't to say that you don't wrestle with those too—I'm just probably quite able to write a configure.ac in an afternoon despite the fact that I've not done it for ten years.

LinApple's build isn't so complex that it really requires it, but if we want things to behave properly UNIXy, it doesn't get any more (pathologically) UNIX than autoconf at least.  ;)